Updated chefspecs + provider bugfix (Fixes #519)

brc opened the pull request:
There were 3 bugs fixed in this commit (one of them was introduced by
me in my refactor):
1.  node attribute `admin_groups` was being checked instead of `groups`
   in `find_dsh_group_members()`
2.  the node attribute branch for the dsh group was being overwritten
   in `configure_pubkey()`
3.  the `$HOME/.dsh` and `$HOME/.dsh/group` directories were being created
   with root ownership (instead of `user_admin` attribute value)

removed `create_dsh_information()` because everything about it was wrong.

corrected the chefspec examples (including a stub) to expect different
values, which correspond with the newly-refactored code.

rcbjenkins commented:
Running GATE test(s): http://build.monkeypuppetlabs.com:8080/job/GIT-Pull-Request/467/console

rcbjenkins commented:
Gate Failed - The patch https://github.com/rcbops-cookbooks/dsh/pull/31.diff could not be applied to grizzly of https://github.com/rcbops/chef-cookbooks.  Please see http://build.monkeypuppetlabs.com:8080/job/GIT-Pull-Request/467/consoleFull for more details.

brc committed:
Updated chefspecs + provider bugfix (Fixes #519)

There were 2 bugs fixed in this commit (hughsaunders fixed the bug I
introduced in my refactor):

1.  the node attribute branch for the dsh group was being overwritten
    in configure_pubkey()
2.  the $HOME/.dsh and $HOME/.dsh/group directories were being created
    with root ownership (instead of `user_admin' attribute value)

removed create_dsh_information() because everything about it was wrong.

corrected the chefspec examples (including a stub) to expect different
values, which correspond with the newly-refactored code.

rcbjenkins commented:
Running GATE test(s): http://build.monkeypuppetlabs.com:8080/job/GIT-Pull-Request/468/console

rcbjenkins commented:
Gate: Submodule Foodcritic: SUCCESS
- http://build.monkeypuppetlabs.com:8080/job/gate-submodule-foodcritic/431/consoleFull

rcbjenkins commented:
Gate: Submodule syntax: SUCCESS
- http://build.monkeypuppetlabs.com:8080/job/gate-submodule-syntax/436/consoleFull

rcbjenkins commented:
Gate: Submodule Tailor (non-voting): WARNING
- http://build.monkeypuppetlabs.com:8080/job/gate-submodule-tailor/382/consoleFull

rcbjenkins commented:
Starting Matrix Gate Job: (jenkins-precise-v2), console URL:  http://build.monkeypuppetlabs.com:8080/job/gate-nova-matrix/BUILD_TYPE=mini-ha,INSTANCE_IMAGE=jenkins-precise-v2,label=nova/622/consoleFull

rcbjenkins commented:
Starting Matrix Gate Job: (jenkins-centos-v2), console URL:  http://build.monkeypuppetlabs.com:8080/job/gate-nova-matrix/BUILD_TYPE=mini-ha,INSTANCE_IMAGE=jenkins-centos-v2,label=nova/622/consoleFull

rcbjenkins commented:
Gate:  Nova AIO (jenkins-centos-v2): SUCCESS
- http://build.monkeypuppetlabs.com:8080/job/gate-nova-matrix/BUILD_TYPE=mini-ha,INSTANCE_IMAGE=jenkins-centos-v2,label=nova/622/consoleFull

rcbjenkins commented:
Gate:  Nova AIO (jenkins-precise-v2): SUCCESS
- http://build.monkeypuppetlabs.com:8080/job/gate-nova-matrix/BUILD_TYPE=mini-ha,INSTANCE_IMAGE=jenkins-precise-v2,label=nova/622/consoleFull
